### ISSUE TITLE
Stop propagation not working for mousemove events of canvas layers 

### ISSUE THREAD

**Issue by Trufi** [Leaflet/Leaflet#5464]
## How to reproduce

- Leaflet version I'm using: 1.03
- Browser (with version) I'm using: Chrome 58
- OS/Platform (with version) I'm using: macOS 10.12
- Add a vector canvas layer on the map (`L.Circle` for example)
- Add a `mousemove` event listener on the layer with stop propagation (`L.DomEvent.stop`)
- Add a `mousemove` listener on the map
- Move the cursor over the vector layer

## What behaviour I'm expecting and which behaviour I'm seeing
I'm expect that the layer listener will stop the event propagation and the map listener won't be called. But when you move the cursor over the vector layer both listeners will be called.

## Minimal example reproducing the issue

- [x] this example is as simple as possible
- [x] this example does not rely on any third party code

Example: http://playground-leaflet.rhcloud.com/paqi/1/edit?html,console,output

`L.DomEvent.stop` not always working, because the listener of the layer catch DOM events with a throttling https://github.com/Leaflet/Leaflet/blob/master/src/layer/vector/Canvas.js#L63

I can make a PR and remove the throttling, but what do you think about the performance degradation in this case? The throttling was did at https://github.com/Leaflet/Leaflet/pull/3547

**Comment by perliedman** [comment]
Hi,
nice analysis. The throttling is indeed problematic in this scenario.

I would however not recommend just throwing out the throttling: as mentioned in the motivation for #3547, this is an important performance optimization, especially when you have a large number of features.

I would argue that it's actually better to make the throttling `stop` all throttled events instead. This has the downside that `mousemove` is now throttled for the map as well, but I think that is a minor issue in practice.

Also, it could be reasonable to make the throttling an option, since the current hardcoded value of `32` might be high for some use cases, but too low for other.

**Comment by Trufi** [comment]
> I would argue that it's actually better to make the throttling stop all throttled events instead. This has the downside that mousemove is now throttled for the map as well, but I think that is a minor issue in practice.

Do I understand correctly that it will work like this:
1. `event` is caught by the throttle wrapper of the listener
2. we stop the event, e.g. `event.stopPropagation()`
3. wait until the listener isn't really called
4. if the event was stopped in the listener, do nothing, else dispatch the event to other listeners

How do we unpause the event and dispatch it to others?
It may looks like `event.target.parentElement.dispatchEvent(event)`, but I'm not sure.

**Comment by perliedman** [comment]
Yeah, the problem is that the throttling function, as implemented now, will just ignore events that occur in too rapid succession, while we want it to first `stopPropagation`, and _then_ ignore. That would require some changes to how throttling is done.
